index.bs

```diff
@@ -1450,6 +1451,47 @@ that are returned to the caller when a new credential is created, or a new asser
         [=extension identifier=] → [=client extension output=] entries produced by the extension's
         [=client extension processing=].
 
+    :   {{PublicKeyCredential/toJSON()}}
+    ::  This operation returns {{RegistrationResponseJSON}} or {{AuthenticationResponseJSON}}, a JSON-serialized representation of the instance of
+        {{PublicKeyCredential}} suitable for submission to a [=[RP]=] server as a payload with `application/json` mimetype. The [=client=] is in
+        charge of encoding any {{ArrayBuffer}} values using [=base64url encoding=] to generate strings compatible with JSON syntax. {{ArrayBuffer}}
+        values to be encoded are identified according to the detected type of the instance's `response` property.
+
+        <xmp class="idl">
+            typedef DOMString Base64URLString;
+            typedef (RegistrationResponseJSON or AuthenticationResponseJSON) PublicKeyCredentialJSON;
+
+            dictionary RegistrationResponseJSON {
+                Base64URLString id;
+                Base64URLString rawId;
+                AuthenticatorAttestationResponseJSON response;
+                DOMString type;
+            };
+
+            dictionary AuthenticatorAttestationResponseJSON {
+                Base64URLString clientDataJSON;
+                Base64URLString attestationObject;
+            };
+
+            dictionary AuthenticationResponseJSON {
+                Base64URLString id;
+                Base64URLString rawId;
+                AuthenticatorAssertionResponseJSON response;
+                DOMString type;
+            };
+
+            dictionary AuthenticatorAssertionResponseJSON {
+                Base64URLString clientDataJSON;
+                Base64URLString authenticatorData;
+                Base64URLString signature;
+                Base64URLString? userHandle;
+            };
+        </xmp>
+
+        <div class="note">
+        Note: This method is intended to make WebAuthn API consumption tenable without requiring the use of third-party libraries.
```

Comment:
Might be better to clarify that it's about preventing client side JS needing to "muck about" with the internals of the types instead I think.

Reply:
Agreed. I also think this note isn't really needed and could just be deleted if we want.

Reply:
These notes have been removed completely.